pageserver/compaction/src/compact_tiered.rs

```diff
@@ -550,16 +548,13 @@ where
         let mut new_jobs = Vec::new();
 
         // Slide a window through the keyspace
-        let mut key_accum = std::pin::pin!(accum_key_values(key_value_stream));
+        let mut key_accum =
+            std::pin::pin!(accum_key_values(key_value_stream, self.target_file_size));
         let mut all_in_window: bool = false;
         let mut window = Window::new();
-        loop {
-            if all_in_window && window.elems.is_empty() {
-                // All done!
-                break;
-            }
-            if let Some(key_range) = window.choose_next_delta(self.target_file_size, !all_in_window)
-            {
+
+        let drain_window = |window: &mut Window<_>, new_jobs: &mut Vec<_>, has_more: bool| {
```

Comment:
has_more is the param passed to the `choose_next_delta` function. it's the negation of the `all_in_window` param. In general, I think we should unify both because I got confused myself when I was thinking about different cases and where I had to do the negation in my head. So either have `has_more` everywhere or `all_in_window` everywhere.